src/test/kotlin/org/pkl/lsp/LSPTestBase.kt

```diff
@@ -168,35 +165,14 @@ abstract class LSPTestBase {
     modules[uri]?.let {
       return it
     }
-    return when (uri.scheme) {
-      "pkl" -> fakeProject.stdlib.getModule(uri.schemeSpecificPart)!!.also { modules[uri] = it }
-      "file" -> {
-        val path = uri.toPath()
-        parseAndStoreModule(path.readText(), path)
-      }
-      "jar" -> {
-        val path = uri.toPath()
-        parseAndStoreJar(path.readText(), path)
-      }
-      else -> throw IllegalArgumentException("Received completion for unknown module: $uri")
-    }
+    val file = fakeProject.virtualFileManager.get(uri)!!
+    return file.getModule().get()!!.also { modules[uri] = it }
   }
 
   private fun resolveToRealUri(uri: String): URI =
     if (uri.startsWith("file:/") && !uri.startsWith("file:///")) Path.of(URI(uri)).toUri()
     else fakeProject.virtualFileManager.get(URI(uri))!!.uri
 
-  private fun parseAndStoreModule(contents: String, path: Path): PklModule {
-    val moduleCtx = parser.parseModule(contents)
-    return PklModuleImpl(moduleCtx, FsFile(path, fakeProject)).also { modules[path.toUri()] = it }
-  }
-
-  private fun parseAndStoreJar(contents: String, path: Path): PklModule {
-    val moduleCtx = parser.parseModule(contents)
-    val uri = path.toUri()
-    return PklModuleImpl(moduleCtx, JarFile(path, uri, fakeProject)).also { modules[uri] = it }
-  }
-
```

Comment:
Since virtual files now handle their own parsing, we don't need this logic anymore.